sys/lora_serialization/lora_serialization.c

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (C) 2018 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @brief       Lora-serialization format implementation
+ *
+ * @author      Leandro Lanzieri <leandro.lanzieri@haw-hamburg.de>
+ * @author      Jose Ignacio Alamos <jose.alamos@haw-hamburg.de>
+ */
+
+#include "byteorder.h"
+#include "assert.h"
+#include "lora_serialization.h"
+
+void lora_serialization_reset(lora_serialization_t *serialization)
+{
+    memset(serialization->buffer, 0, LORA_SERIALIZATION_MAX_BUFFER_SIZE);
+    serialization->cursor = 0;
+}
+
+void lora_serialization_write_unix_time(lora_serialization_t *serialization,
+                                        uint32_t unixtime)
+{
+    le_uint32_t *value = (le_uint32_t *)
+                         (serialization->buffer + serialization->cursor);
+
+    assert((serialization->cursor + LORA_SERIALIZATION_UNIX_TIME_SIZE) <
+           LORA_SERIALIZATION_MAX_BUFFER_SIZE);
+
+    *value = byteorder_btoll(byteorder_htonl(unixtime));
+    serialization->cursor += LORA_SERIALIZATION_UNIX_TIME_SIZE;
+}
+
+void lora_serialization_write_coordinates(lora_serialization_t *serialization,
+                                          double latitude, double longitude)
+{
+    int32_t lat = latitude * 1e6;
+    int32_t lng = longitude * 1e6;
```

Comment:
You could avoid float operations by providing latitude and longitude as e.g. `latitude`, `latitude_frac` integers.